Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Promact.Trappist.DomainModel.Models.Category;
+using Promact.Trappist.Repository.Category;
+
+namespace Promact.Trappist.Core.Controllers
+{
+    [Route("api/Category")]
+    public class CategoryController : Controller
+    {
+        private ICategoryRepository _categoriesRepository;
+        public CategoryController(ICategoryRepository categoriesRepository)
+        {
+            _categoriesRepository = categoriesRepository;
+        }
+
+        #region post Method 
+        [HttpPost]
+        /// <summary>
+        /// Post Method 
+        /// Will Add a Catagory Name in Category Model
+        ///</summary>
+        /// <param name="category">Object of  class Category</param>
+        /// <returns>object of the class </returns>
+        public IActionResult CatagoryAdd([FromBody] Category category)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            _categoriesRepository.AddCategory(category);
+            return NoContent();
+        }
+        #endregion
+
+        #region PutMethod
+        /// <summary>
+        /// Put Method
+        /// Will Edit a Existing Category from Category Table
+        /// </summary>
+        /// <param name="Id">Id is the primary key of Category Model</param>
+        /// <param name="catagory">Object of  class Category</param>
+        /// <returns>object of the class</returns>
+        [HttpPut("{id}")]
+        public IActionResult CategoryEdit(int Id, [FromBody] Category category)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            var promise = _categoriesRepository.Getcategory(Id);
```

Comment:
there is no point that id willnot in database because we have an option to edit a existing Category name